torch/onnx/_internal/exporters/fx_symbolic.py

```diff
@@ -165,59 +165,54 @@ def _module_expansion_symbolic_trace(
-    # Basically, we want
-    #  ModeoProto.graph.input =
-    #   [input_0, input_1, ..., input_n, weight_0, weight_1, ..., weight_m]
-    # and we don't want
-    #  ModeoProto.graph.input =
-    #   [input_0, weight_0, input_1, weight_1, ..., input_n, weight_0, weight_1, ..., weight_m]
-    graph_module = passes.MovePlaceholderToFront(graph_module).run()
-    # To save memory, move get_attr to input so that the generated model doesn't
-    # have weigh tensors. "replaced_attrs" are a tuple of replaced weight tensors.
-    replace_get_attr_with_placeholder_pass = passes.ReplaceGetAttrWithPlaceholder(
-        graph_module
-    )
-    graph_module = replace_get_attr_with_placeholder_pass.run()
-    replaced_attrs = replace_get_attr_with_placeholder_pass.replaced_attrs
-    # Move all newly created placeholder nodes to the front of the graph.
-    graph_module = passes.MovePlaceholderToFront(graph_module).run()
-    # Finalize the graph editing.
-    graph_module.recompile()
-    return (
-        exporter._export(
-            graph_module,
-            (*bound_args, *replaced_attrs),
-            opset_version=opset_version,
-            decomposition_table=decomposition_table,
-            use_binary_format=use_binary_format,
-            op_level_debug=op_level_debug,
-            enable_dynamic_axes=enable_dynamic_axes,
-        ),
-        graph_module,
-        bound_args,
-        replaced_attrs,
-    )
+        # Make sure all placeholder nodes are executed before get_attr nodes.
+        # Otherwise, inputs can interleave with initializers in the final ModeoProto.graph.input.
+        # Basically, we want
+        #  ModeoProto.graph.input =
+        #   [input_0, input_1, ..., input_n, weight_0, weight_1, ..., weight_m]
+        # and we don't want
+        #  ModeoProto.graph.input =
+        #   [input_0, weight_0, input_1, weight_1, ..., input_n, weight_0, weight_1, ..., weight_m]
+        graph_module = passes.MovePlaceholderToFront(graph_module).run()
+        # To save memory, move get_attr to input so that the generated model doesn't
+        # have weigh tensors. "replaced_attrs" are a tuple of replaced weight tensors.
+        replace_get_attr_with_placeholder_pass = passes.ReplaceGetAttrWithPlaceholder(
+            graph_module
+        )
+        graph_module = replace_get_attr_with_placeholder_pass.run()
+        replaced_attrs = replace_get_attr_with_placeholder_pass.replaced_attrs
+        # Move all newly created placeholder nodes to the front of the graph.
+        graph_module = passes.MovePlaceholderToFront(graph_module).run()
+        # Finalize the graph editing.
+        graph_module.recompile()
+
+        export_output = self.export_fx_to_onnx(
+            graph_module, *bound_args, *replaced_attrs
+        )
+
+        return FXSymbolicTraceExportOutput(
+            model_proto=export_output.model_proto,
+            fx_graph_module=graph_module,
```

Comment:
Yeah I think this can go away considering tests don't actually leverage the extras. I did this to capture the original intent of the symbolic exporting function.